File: atcoder/abc208C_fair_candy_distribution.hs
-- uninhm
-- https://atcoder.jp/contests/abc208/tasks/abc208_c
-- implementation

import Data.List (sort)

getInts :: IO [Int]
getInts = map read . words <$> getLine

solve n k a =
  flip map a $ \i ->
    if i < x then
      y + 1
    else
      y
    where x = sort a !! mod k n
          y = k `div` n

main = do
  [n, k] <- getInts
  a <- getInts
  putStr . unlines . map show $ solve n k a

File: atcoder/abc208C_fair_candy_distribution.py
# uninhm
# https://atcoder.jp/contests/abc208/tasks/abc208_c
# implementation

n, k = map(int, input().split())
a = list(map(int, input().split()))

x = sorted(a)[k%n]
for i in a:
    if i < x:
        print(k//n+1)
    else:
        print(k//n)
